chia/cmds/coin_funcs.py

```diff
@@ -27,23 +27,22 @@ async def async_list(args: Dict[str, Any], wallet_client: WalletRpcClient, finge
     if not await wallet_client.get_synced():
         print("Wallet not synced. Please wait.")
         return
-    min_coin_amount = Decimal(args["min_coin_amount"])
-    max_coin_amount = Decimal(args["max_coin_amount"])
+    min_coin_amount: uint64 = uint64(int(Decimal(args["min_coin_amount"]) * mojo_per_unit))
+    max_coin_amount: uint64 = uint64(int(Decimal(args["max_coin_amount"]) * mojo_per_unit))
     excluded_coin_ids = args["excluded_coin_ids"]
-    excluded_amounts = args["excluded_amounts"]
+    excluded_amounts: List[uint64] = [
+        uint64(int(Decimal(amount) * mojo_per_unit)) for amount in args["excluded_amounts"]
+    ]
     addr_prefix = args["addr_prefix"]
     show_unconfirmed = args["show_unconfirmed"]
     paginate = args["paginate"]
     if paginate is None:
         paginate = sys.stdout.isatty()
-    final_min_coin_amount: uint64 = uint64(int(min_coin_amount * mojo_per_unit))
-    final_max_coin_amount: uint64 = uint64(int(max_coin_amount * mojo_per_unit))
-    final_excluded_amounts: List[uint64] = [uint64(int(Decimal(amount) * mojo_per_unit)) for amount in excluded_amounts]
     conf_coins, unconfirmed_removals, unconfirmed_additions = await wallet_client.get_spendable_coins(
         wallet_id=wallet_id,
-        max_coin_amount=final_max_coin_amount,
-        min_coin_amount=final_min_coin_amount,
-        excluded_amounts=final_excluded_amounts,
+        max_coin_amount=max_coin_amount,
+        min_coin_amount=min_coin_amount,
+        excluded_amounts=excluded_amounts,
```

Comment:
all of these are now specified in mojos, whereas they used to be specified in whatever CAT the wallet manages.

This is changing the interface, and behavior, of a public class. You can't update all 3rd party uses of this. I don't think this is safe. Am I missing something?

Reply:
> all of these are now specified in mojos, whereas they used to be specified in whatever CAT the wallet manages.
> 
> This is changing the interface, and behavior, of a public class. You can't update all 3rd party uses of this. I don't think this is safe. Am I missing something?

I'm a bit confused, the internals of a cli command should never have any 3rd party uses & all i did was simplify the variables

Reply:
perhaps I'm misreading this, but `wallet_client` really sounds like our client library that speaks the RPC protocol. The library that we would expect 3rd party developers to use to communicate with a wallet. Is it not?

Reply:
> perhaps I'm misreading this, but `wallet_client` really sounds like our client library that speaks the RPC protocol. The library that we would expect 3rd party developers to use to communicate with a wallet. Is it not?

yeah it is, however I changed the variables above, so final_excluded_amounts on main, is the same as excluded_amounts on this pr.

Reply:
all i did was change the internals, but what its giving to the rpc is the same thing.